sift/src/main/java/siftscience/android/Sift.java

```diff
@@ -256,213 +222,368 @@ static String getQueueIdentifier(String key) {
+                }
+                editor.apply();
+            } catch (JsonParseException e) {
+                Log.e(TAG, "Encountered JsonProcessingException in save", e);
+            }
+        }
+    }
+
+    /**
+     * Restores all of the Sift instance state from disk.
+     */
+    private class UnarchiveTask implements Runnable {
+        @Override
+        public void run() {
+            String archive;
+
+            // Unarchive Sift config
+            archive = archives.getString(ArchiveKey.CONFIG.key, null);
+            config = unarchiveConfig(archive);
+            Log.d(TAG, String.format("Unarchived Sift config: \"%s\"", archive));
+
+            // Unarchive User ID
+            userId = archives.getString(ArchiveKey.USER_ID.key, null);
+            Log.d(TAG, String.format("Unarchived User ID: \"%s\"", userId));
+
+            // Unarchive Queues
+            for (Map.Entry<String, ?> entry : archives.getAll().entrySet()) {
+                String identifier = ArchiveKey.getQueueIdentifier(entry.getKey());
+                if (identifier != null) {
+                    archive = (String) entry.getValue();
+                    Queue queue = new Queue(archive, userIdProvider, uploadRequester);
+                    Log.d(TAG, String.format("Unarchived \"%s\" Queue", identifier));
+                    queues.put(identifier, queue);
+                }
+            }
+
+            if (!queues.containsKey(DEVICE_PROPERTIES_QUEUE_IDENTIFIER)) {
+                createQueue(DEVICE_PROPERTIES_QUEUE_IDENTIFIER, DEVICE_PROPERTIES_QUEUE_CONFIG);
+            }
+
+            if (!queues.containsKey(APP_STATE_QUEUE_IDENTIFIER)) {
+                createQueue(APP_STATE_QUEUE_IDENTIFIER, APP_STATE_QUEUE_CONFIG);
+            }
+        }
+    }
+
+    private class SetUserIdTask implements Runnable {
+        private Sift sift;
+        private String userId;
+
+        SetUserIdTask(Sift sift, String userId) {
+            this.sift = sift;
+            this.userId = userId;
+        }
+
+        @Override
+        public void run() {
+            this.sift.userId = this.userId;
+        }
+    }
+
+    private class SetConfigTask implements Runnable {
+        private Sift sift;
+        private Sift.Config config;
+
+        SetConfigTask(Sift sift, Config config) {
+            this.sift = sift;
+            this.config = config;
+        }
+
+        @Override
+        public void run() {
+            this.sift.config = this.config;
+        }
+    }
+
+    /**
+     * Appends an event to the specified queue.
+     */
+    private class AppendTask implements Runnable {
+        private String queueIdentifier;
+        private MobileEventJson event;
+
+        AppendTask(String queueIdentifier, MobileEventJson event) {
+            this.queueIdentifier = queueIdentifier;
+            this.event = event;
+        }
+
+        @Override
+        public void run() {
+            getQueue(this.queueIdentifier).append(this.event);
```

Comment:
`getQueue(this.queueIdentifier)` could be null

Reply:
check added